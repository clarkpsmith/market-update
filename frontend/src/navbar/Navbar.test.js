import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NavBar from "./NavBar";

import { Provider } from "react-redux";

import configureMockStore from "redux-mock-store";
const mockStore = configureMockStore();

const store = mockStore({
  currentUser: {
    firstName: "test",
    lastName: "user",
    email: "testuser@gmail.com",
    username: "testuser",
    watchlist: ["AAPL"],
  },
});

it("renders without crashing", () => {
  render(
    <Provider store={store}>
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    </Provider>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("displays sign up and log in if theres no current user", () => {
  const store = mockStore({
    currentUser: {},
  });
  const { queryByText } = render(
    <Provider store={store}>
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    </Provider>
  );
  expect(queryByText("Sign Up")).toBeInTheDocument();
  expect(queryByText("Login")).toBeInTheDocument();
});
