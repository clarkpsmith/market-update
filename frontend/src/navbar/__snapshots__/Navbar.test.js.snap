// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="NavBar"
  >
    <nav
      class="navbar navbar navbar-expand-md navbar-dark"
    >
      <a
        aria-current="page"
        class="Navbar-brand active"
        href="/"
      >
        <span
          class="Navbar-logo"
        >
          <img
            alt="stock-market-icon"
            src="stock-market-icon.png"
            width="25px"
          />
        </span>
        Market Update
      </a>
      <button
        aria-label="Toggle navigation"
        class="mr-2 Navbar-toggler-btn navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
      >
        <ul
          class="ml-auto navbar-nav"
        >
          <li
            class="nav-item"
          >
            <a
              class="navlinks"
              href="/news"
            >
              News
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="navlinks"
              href="/search"
            >
              Search
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="navlinks"
              href="/watchlist"
            >
              Watchlist
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="navlinks"
              href="/profile"
            >
              Profile
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="navlinks-logout active"
              href="/"
            >
              Log Out
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
